plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer-v2.ts

```diff
@@ -185,6 +192,19 @@ export class SessionRecordingIngesterV2 {
             return
         }
 
+        // Check that we are not below the high water mark for this partition (another consumer may have flushed further than us when revoking)
+        if (await this.offsetHighWaterMarker.isBelowHighWaterMark(event.metadata, HIGH_WATERMARK_KEY, offset)) {
+            eventDroppedCounter
+                .labels({
+                    event_type: 'session_recordings_blob_ingestion',
+                    drop_cause: 'high_water_mark_partition',
+                })
+                .inc()
+
+            highWaterMarkSpan?.finish()
+            return
+        }
+
```

Comment:
So, thinking this through we have a per session highwater mark and a per topic/partition highwater mark. 

We don't need to worry about orphaned sessions because we don't chunk and so if a commit makes it to the per TP highwatermark it must have been written to S3

If we discovered that wasn't the case then we fix that not the highwater mark tracking

Cool 👍

Reply:
I don't think we need to worry about this no.

The TP water mark is basically mirroring what we commit to kafka and is just for us to handle the exceptional case where we start consuming on one pod whilst we are still flushing on another.